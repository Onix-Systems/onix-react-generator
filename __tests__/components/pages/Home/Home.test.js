import React from 'react';
import fetchMock from 'jest-fetch-mock';
import { render } from '../../../settings/test-utils';
import HomePage from '../../../../src/pages/Home/Home';

describe('<Home />', () => {
  let wrapper;
  beforeEach(() => {
    fetchMock.mockResponseOnce(JSON.stringify('mock data'));
    fetchMock.mockRejectOnce(JSON.stringify('mock error'));
    wrapper = render(<HomePage />);
  });

  test('should render Home component', (done) => {
    const { asFragment } = wrapper;
    expect(asFragment(<HomePage />)).toMatchSnapshot();
    done();
  });

  test('should call componentDidMount', (done) => {
    const mockComponentDidMount = jest.spyOn(HomePage.WrappedComponent.prototype, 'componentDidMount');
    render(<HomePage />);
    expect(mockComponentDidMount).toHaveBeenCalledTimes(1);
    done();
  });
});
