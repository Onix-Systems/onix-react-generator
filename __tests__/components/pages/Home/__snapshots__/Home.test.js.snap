// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render Home component 1`] = `
<DocumentFragment>
  <div>
    <p
      class="appIntro"
    >
      Home page
    </p>
    <p>
      mock data
    </p>
    <button
      class="updateJoke"
      data-testid="button"
      type="button"
    >
      New joke
    </button>
  </div>
</DocumentFragment>
`;
